### PR TITLE
Adding support for `func subscribe` for creating mutiple triggers, based on event filters

### PR DESCRIPTION
**Pull request by matzew**
Little demo: https://youtu.be/O6M7mfR6JfM 

# Changes

- - :gift: Add new `subscribe` command to create Knative Triggers for event routing

/kind <kind>

Fixes #

**Release Note**

```release-note
kn func subscribe will allow you to create Knative Eventing Triggers for improved event processing for kn func
```

**Docs**

```docs

```

### CODE REVIEW

File: pkg/knative/deployer.go [hunk 1]
```diff
@@ -189,6 +196,40 @@ func (d *Deployer) Deploy(ctx context.Context, f fn.Function) (fn.DeploymentResu
...
+			for i, sub := range f.Subscription {
+
+				// create the filter:
+				attributes := make(map[string]string)
+				for key, value := range sub.Filters {
+					attributes[key] = value
+				}
+
+				err = eventingClient.CreateTrigger(ctx, &eventingv1.Trigger{
+					ObjectMeta: metav1.ObjectMeta{
```
Comment (lkingland) [inline pkg/knative/deployer.go:208]: I love how straightforward the API call is here

**Review by lkingland — COMMENTED** [review]
Looking good so far!

**Comment by matzew** [comment]
/hold

the filtering needs to be a bit different

**Comment by jrangelramos** [comment]
Hi @matzew . If possible, could you add some test pls?

**Comment by lkingland** [comment]
> Hi @matzew . If possible, could you add some test pls?

We _do_ install Eventing in our test cluster, so an integration test in https://github.com/knative/func/blob/main/pkg/functions/client_int_test.go might be fairly straightforward

File: cmd/root.go [hunk 2]
```diff
@@ -76,6 +76,7 @@ Learn more about Knative at: https://knative.dev`, cfg.Name),
 				NewDeployCmd(newClient),
 				NewDeleteCmd(newClient),
 				NewListCmd(newClient),
+				NewSubscribeCmd(),
```
Comment (lkingland) [inline cmd/root.go:79]: For testing support, this should pass through the `newClient` client factory; but I don't think that should hold up the PR.

**Review by lkingland — APPROVED** [review]

File: cmd/root.go (hunk 2 above)
Comment (matzew) [reply cmd/root.go:79]: I will add additional PR. thanks for the review on this

**Pull request merged**